core/algorithms/onpolicy_sync/engine.py

```diff
@@ -624,13 +624,18 @@ def __init__(
         self.last_log: Optional[int] = None
         self.last_save: Optional[int] = None
 
-    def advance_seed(self, seed: Optional[int]) -> Optional[int]:
+    def advance_seed(
+        self, seed: Optional[int], call_worker_seeds=True
+    ) -> Optional[int]:
         if seed is None:
             return seed
         seed = (seed ^ (self.training_pipeline.total_steps + 1)) % (
             2 ** 31 - 1
         )  # same seed for all workers
 
+        if not call_worker_seeds:
+            return seed
+
```

Comment:
I like the idea of having something like this but can we rename `call_worker_seeds` to something like `return_same_seed_per_worker` and then rather than immediately returning seed here we could instead change 
```python
if self.mode == "train" or self.mode == "test":
```
to
```python
if (not return_same_seed_per_worker) and (self.mode == "train" or self.mode == "test"):
```
below? I like this slightly better as things will be a bit more consistent I think.